sched/semaphore/spinlock.c

```diff
@@ -384,4 +384,176 @@ void spin_clrbit(FAR volatile cpu_set_t *set, unsigned int cpu,
 }
 #endif
 
+/****************************************************************************
+ * Name: fair_spinlock_list_init
+ *
+ * Description:
+ *   Initialize the link linsk of fair spinlock.
+ *   If there is any spinlock node, the node will get lock.
+ *
+ * Input Parameters:
+ *   lock_list - A reference to the spinlock list.
+ *
+ * Returned Value:
+ *   None
+ *
+ * Assumptions:
+ *   None.
+ *
+ ****************************************************************************/
+
+void fair_spinlock_list_init(fair_spinlock_list_t *lock_list)
+{
+  lock_list->lock = SP_UNLOCKED;
+  list_initialize(&lock_list->list);
+}
+
+/****************************************************************************
+ * Name: fair_spinlock_init
+ *
+ * Description:
+ *   Initialize the fair spinlock. Tje lock state is set as SP_LOCK util
+ *   some give lock to holder.
+ *
+ * Input Parameters:
+ *   lock - A reference to the spinlock object to lock.
+ *
+ * Returned Value:
+ *   None
+ *
+ * Assumptions:
+ *   Not running at the interrupt level.
+ *
+ ****************************************************************************/
+
+void fair_spinlock_init(fair_spinlock_t *lock)
+{
+  lock->lock = SP_LOCKED;
```

Comment:
why init to the lock status?

Reply:
Default there is a guy get this lock.
I will explain more detail on the last response.
Hope you can give some suggestions.